GLTF/src/GLTFAsset.cpp

```diff
@@ -440,6 +493,55 @@ GLTF::BufferView* packAccessorsForTargetByteStride(std::vector<GLTF::Accessor*>
 	return bufferView;
 }
 
+bool GLTF::Asset::compressPrimitives(GLTF::Options* options) {
+	int totalPrimitives = 0;
+	for (GLTF::Primitive* primitive : getAllPrimitives()) {
+		totalPrimitives++;
+		auto dracoExtensionPtr = primitive->extensions.find("KHR_draco_mesh_compression");
+		if (dracoExtensionPtr == primitive->extensions.end()) {
+			// No extension exists.
+			continue;
+		}
+		GLTF::DracoExtension* dracoExtension = (GLTF::DracoExtension*)dracoExtensionPtr->second;
+		draco::Mesh *dracoMesh = dracoExtension->dracoMesh.get();
+		if (!dracoMesh) {
+			continue;
+		}
+
+		// Compress the mesh
+		// Setup encoder options.
+		draco::Encoder encoder;
+		int pos_quantization_bits = options->position_quantization_bits;
+		int tex_coords_quantization_bits = options->texcoord_quantization_bits;
+		int normals_quantization_bits = options->normal_quantization_bits;
+		int color_quantization_bits = options->color_quantization_bits;
+		// Used for compressing joint indices and joint weights.
+		int generic_quantization_bits = options->joint_quantization_bits;
+
+		encoder.SetAttributeQuantization(draco::GeometryAttribute::POSITION, pos_quantization_bits);
+		encoder.SetAttributeQuantization(draco::GeometryAttribute::TEX_COORD, tex_coords_quantization_bits);
+		encoder.SetAttributeQuantization(draco::GeometryAttribute::NORMAL, normals_quantization_bits);
+		encoder.SetAttributeQuantization(draco::GeometryAttribute::COLOR, color_quantization_bits);
+		encoder.SetAttributeQuantization(draco::GeometryAttribute::GENERIC, generic_quantization_bits);
+
+		draco::EncoderBuffer buffer;
+		const draco::Status status = encoder.EncodeMeshToBuffer(*dracoMesh, &buffer);
+		if (!status.ok()) {
+			std::cerr << "Error: Encode mesh.\n";
```

Comment:
Is there more information available from draco here? It would be nice to get a more descriptive error if something goes wrong.